manifests.go

```diff
@@ -0,0 +1,49 @@
+package distribution
+
+import "github.com/docker/distribution/digest"
+
+// Manifest specifies a registry object with a list of dependencies.
+type Manifest interface {
+	Blob // io access to serialized content.
+
+	// Dependencies returns a list of object on which this manifest depends.
+	// The dependencies are strictly ordered from base to head. Typically,
+	// these are layers but their interpretation is application specific.
+	Dependencies() []Descriptor
```

Comment:
So if I want to "walk" the manifest tree, I must:

1. Iterate on its `Dependencies()`
2. Call `BlobService.Get(dependency.Digest)` to fetch the dependency blob
3. Switch on `dependencyBlob.Descriptor().MediaType` to determine what it is? Or `manifest ok := dependencyBlob.(Manifest)`?

To rephrase: I don't clearly see if what I fetch will be base blobs which `MediaType` I must examine, or if the library will do the job of recognizing a few mediaTypes (such as the one(s?) that identify a manifest) to instantiate `Blob`-descendent types (such as a `Manifest`).